src/features/income-management/pages/lease-management/components/generate-modal-components/generate-templates/BSTemplate.tsx

```
@@ -0,0 +1,124 @@
+import BaseModal from "@/features/income-management/components/BaseModal";
+import { Grid, Text, Divider } from "@mantine/core";
+import Table from "@/components/ui/table/components/Table";
+import { generateModalConfigs } from "../../../config/generate-modal-config";
+import { getTitle } from "../../../config/generate-modal-config";
+
+interface BSTemplateProps {
+  data?: any;
+  onClose: () => void;
+  viewType: string;
+}
+
+const BSTemplate: React.FC<BSTemplateProps> = ({ data, onClose, viewType }) => {
+  const config = generateModalConfigs[viewType];
+
+  if (!config) return null;
```

what's config?

consider throwing an error instead of returning null to highlight the need for the config value

```typescript
if (!config) throw new Error("Config is required")

```